plugins/org.python.pydev.core/pysrc/pydev_coverage.py

```diff
@@ -22,34 +41,45 @@ def execute():
                 s = input()
             s = s.replace('\r', '')
             s = s.replace('\n', '')
-            files = s.split('|')
-            files = [v for v in files if len(v) > 0]
+            all_files = s.split('|')
+
+            files = [v for v in all_files if is_valid_py_file(v)]
```

Comment:
I think you should do a single pass in `all_files` to separate the files/invalid files instead of doing 2 list comprehensions.

Reply:
Thanks for the fast review. Do I understand you right that you have a classic loop in mind?

    files = []
    invalid_files = []
    for v in s.split('|'):
        if is_valid_py_file(v):
            files.append(v)
        else:
            invalid_files.append(v)

Reply:
Yes.